crates/uv-python/fetch-download-metadata.py

```diff
@@ -1,272 +1,389 @@
+        """Parse an indygreg download URL into a PythonDownload object."""
+        # Ex)
+        # https://github.com/indygreg/python-build-standalone/releases/download/20240107/cpython-3.12.1%2B20240107-aarch64-unknown-linux-gnu-lto-full.tar.zst
+        filename = unquote(url.rsplit("/", maxsplit=1)[-1])
+        if filename.endswith(".sha256"):
+            return
+
+        match = self._filename_re.match(filename)
+        if match is None:
+            return
+
+        version, triple = match.groups()
+        if triple.endswith("-full"):
+            triple = triple[:-5]
+
+        match = self._flavor_re.match(triple)
+        if match is not None:
+            triple, flavor = match.groups()
+        else:
+            flavor = None
+        if flavor in self.HIDDEN_FLAVORS:
+            return
+
+        version = Version.from_str(version)
+        triple = self._normalize_triple(triple)
+        if triple is None:
+            return
+
+        return PythonDownload(
+            version=version,
+            triple=triple,
+            flavor=flavor,
+            implementation=self.implementation,
+            filename=filename,
+            url=url,
+        )
+
+    def _normalize_triple(self, triple: str) -> PlatformTriple | None:
+        if "-static" in triple:
+            logging.debug("Skipping %r: static unsupported", triple)
+            return
+
+        triple = self.SPECIAL_TRIPLES.get(triple, triple)
+        pieces = triple.split("-")
+        try:
+            arch = self._normalize_arch(pieces[0])
+            operating_system = self._normalize_os(pieces[2])
+            if pieces[2] == "linux":
+                # On linux, the triple has four segments, the last one is the libc
+                libc = pieces[3]
+            else:
+                libc = "none"
+        except IndexError:
+            logging.debug("Skipping %r: unknown triple", triple)
+            return
+
+        return PlatformTriple(arch, operating_system, libc)
+
+    def _normalize_arch(self, arch: str) -> str:
+        arch = self.ARCH_MAP.get(arch, arch)
+        pieces = arch.split("_")
+        # Strip `_vN` from `x86_64`
+        return "_".join(pieces[:2])
+
+    def _normalize_os(self, os: str) -> str:
+        return os
+
+    def _get_flavor_priority(self, flavor: str) -> int:
+        """
+        Returns the priority of a flavor. Lower is better."""
+        try:
+            pref = self.FLAVOR_PREFERENCES.index(flavor)
+        except ValueError:
+            pref = len(self.FLAVOR_PREFERENCES) + 1
+        return pref
+
+
+def render(downloads: list[PythonDownload]) -> None:
+    """Render `download-metadata.json`."""
+
+    def sort_key(download: PythonDownload) -> tuple:
+        # Sort by implementation, version (latest first), and then by triple.
+        impl_order = [ImplementationName.CPYTHON, ImplementationName.PYPY]
+        return (
+            impl_order.index(download.implementation),
```

Comment:
Is there a specific reason why we use `index` to maintain the order here, instead of relying solely on the lexical order of the `download.implementation` (`cpython` followed by `pypy`) which is our expected order?

Reply:
I do think we want CPython to come before PyPy regardless of lexical ordering.